File: Programming Basics/Programming Basics - C#/Exercises/05. Simple Loops/05. Simple Loops/Odd - Even Position/Equal Pairs.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Odd___Even_Position
{
    class Program
    {
        static void Main(string[] args)
        {
            int n = int.Parse(Console.ReadLine());
            double previousPairSum = 0;
            double currentPairSum = 0;
            double difference = 0;
            double biggestdifference = 0;

            for (int i = 0; i < n; i++)
            {
                double firstNumber = double.Parse(Console.ReadLine());
                double secondNumber = double.Parse(Console.ReadLine());

                if (i == 0)
                {
                    previousPairSum = firstNumber + secondNumber;
                }
                else
                {
                    currentPairSum = firstNumber + secondNumber;
                    difference = Math.Abs(currentPairSum - previousPairSum);
                    if (difference > biggestdifference)
                    {
                        biggestdifference = difference;
                    }
                    previousPairSum = currentPairSum;
                }
            }

            if (biggestdifference == 0)
            {
                Console.WriteLine("Yes, value ={0}", previousPairSum);
            }
            else
            {
                Console.WriteLine("No, maxdiff ={0}", biggestdifference);
            }
        }

    }
}
    

File: Programming Basics/Programming Basics - JavaScript/Exercises/05. Simple Loops/12. Equal Pairs/EqualPairs.js
function equalPairsOrNot(args) {
    let n = Number(args[0]);

    let previousPairSum = 0;
    let currentPairSum = 0;
    let difference = 0;
    let biggestdifference = 0;

    for (let i = 1; i <= n; i++) {
        let firstNumber = Number(args[1]);
        let secondNumber = Number(args[2]);

        if (i == 1) {
            previousPairSum = firstNumber + secondNumber;
        }

        else {
            currentPairSum = firstNumber + secondNumber;
            difference = Math.abs(currentPairSum - previousPairSum);

            if (difference > biggestdifference) {
                biggestdifference = difference;
            }

            previousPairSum = currentPairSum;
        }
    }

    if (biggestdifference == 0) {
        console.log(`Yes, value=${previousPairSum}`);
    }
    else {
        console.log(`No, maxdiff=${biggestdifference}`);
    }
}

equalPairsOrNot(["3", "1", "2", "0", "3", "4", "-1"]);
equalPairsOrNot(["4", "1", "1", "3", "1", "2", "2", "0", "0"]);
equalPairsOrNot(["2", "-1", "0", "0", "-1"]);
equalPairsOrNot(["2", "1", "2", "2", "2"]);